File: 0763-partition-labels/0763-partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        res = []
        arr = [0] * 26
        for i in range(len(s)):
            arr[ord(s[i]) - ord('a')] = i
        i = 0
        j = -1
        index = 0
        while i < len(s):
            if arr[ord(s[i]) - ord('a')] > j:
                j = arr[ord(s[i]) - ord('a')]
            if i == j:
                res.append(i - index + 1)
                index = i + 1
            i += 1
        return res

File: 0763-partition-labels/0763-partition-labels.java
class Solution {
    public List<Integer> partitionLabels(String s) {
        List<Integer> list=new ArrayList<>();
        int arr[]=new int[26];
        for(int i=0; i<s.length(); i++){
            arr[s.charAt(i)-'a']=i;
        }
        int i=0, j=-1, index=0;
        while(i<s.length()){
            if(arr[s.charAt(i)-'a']>j)j=arr[s.charAt(i)-'a'];
            if(i==j){
                list.add(i-index+1);
                index=i+1;
            }
            i++;
        }

        return list;

    }
}